applications/StructuralMechanicsApplication/CMakeLists.txt

```diff
@@ -6,168 +6,28 @@ message("**** configuring KratosStructuralMechanicsApplication ****")
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/point_contact_condition.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/axisym_point_load_condition.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/line_load_condition.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/line_load_condition_2d.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/axisym_line_load_condition_2d.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/surface_load_condition_3d.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/point_moment_condition_3d.cpp
-
-  # Adding conditions for adjoint sensitivity analysis
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_response_functions/adjoint_conditions/adjoint_semi_analytic_point_load_condition.cpp
+## StructuralMechanics Core sources
+file(GLOB_RECURSE KRATOS_STRUCTURAL_MECHANICS_APPLICATION_CORE
+    ${CMAKE_CURRENT_SOURCE_DIR}/structural_mechanics_application.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/structural_mechanics_application_variables.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_utilities/*.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/*.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_constitutive/*.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/*.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_io/*.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_processes/*.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_response_functions/*.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_strategies/*.cpp
 )
 
-## generate variables with the testing sources
+## StructuralMechanics testing sources
 if(${KRATOS_BUILD_TESTING} MATCHES ON)
-  file(GLOB_RECURSE
-    KRATOS_STRUCTURAL_MECHANICS_APPLICATION_TESTING_SOURCES
-    ${CMAKE_CURRENT_SOURCE_DIR}/tests/cpp_tests/*.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/tests/bits/*.cpp
-  )
+    file(GLOB_RECURSE KRATOS_STRUCTURAL_MECHANICS_APPLICATION_TESTING_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/tests/*.cpp)
 endif(${KRATOS_BUILD_TESTING} MATCHES ON)
 
-## generate variables with the sources
-set( KRATOS_STRUCTURAL_MECHANICS_APPLICATION_PYTHON_INTERFACE
-  ## CUSTOM PYTHON
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_python/structural_mechanics_python_application.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_python/add_custom_strategies_to_python.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_python/add_custom_processes_to_python.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_python/add_custom_utilities_to_python.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_python/add_custom_constitutive_laws_to_python.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/custom_python/add_custom_response_functions_to_python.cpp
-)
+## StructuralMechanics python interface sources
+file(GLOB_RECURSE KRATOS_STRUCTURAL_MECHANICS_APPLICATION_PYTHON_INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/custom_python/*.cpp)
 
 add_library(KratosStructuralMechanicsCore SHARED ${KRATOS_STRUCTURAL_MECHANICS_APPLICATION_CORE} ${KRATOS_STRUCTURAL_MECHANICS_APPLICATION_TESTING_SOURCES})
 target_link_libraries(KratosStructuralMechanicsCore PUBLIC KratosCore)
```

Comment:
@roigcarlo should this link public or private?
I think private, thats at least what @jcotela told me

Also Fluid does it private

Reply:
nevermind, I meant the other one